doc/arkode/guide/source/Usage/ASA.rst

```diff
@@ -0,0 +1,159 @@
+
+   * If a user-defined partition integrator was supplied, free the integrator
+     content and call :c:func:`SUNStepper_Destroy` to free the :c:type:`SUNStepper`
+     object.
+
+   * Call :c:func:`ARKodeFree` to free the memory allocated for the
+     SplittingStep integration object.
+
+#. *Free the SUNContext object*
+
+#. *Finalize MPI, if used*
+
+
+
+User Callable Functions
+-----------------------
+
+This section describes ERKStep-specific user-callable functions for performing
+adjoint sensitivity analysis with methods with ERKStep.
+
+.. c:function:: int ERKStepCreateAdjointStepper(void* arkode_mem, N_Vector sf, SUNAdjointStepper* adj_stepper_ptr)
+
+   Creates a :c:type:`SUNAdjointStepper` object compatible with the provided ARKStep instance for
+   integrating the adjoint sensitivity system :eq:`ARKODE_DISCRETE_ADJOINT`.
+
+   :param arkode_mem: a pointer to the ARKStep memory block.
+   :param sf: the sensitivity vector holding the adjoint system terminal condition.
+      This must be an instance of the ManyVector ``N_Vector`` implementation with at
+      least one subvector (depending on if sensitivities to parameters should be computed).
+      The first subvector must be :math:`\partial g_y(y(t_f)) \in \mathbb{R}^N`. If sensitivities to parameters should be computed, then the second subvector must be :math:`g_p(y(t_f), p) \in \mathbb{R}^{N_s}`,
+      otherwise only one subvector should be provided.
+   :param adj_stepper_ptr: the newly created :c:type:`SUNAdjointStepper` object.
+
+   :return:
+      * ``ARK_SUCCESS`` if successful
+      * ``ARK_MEM_FAIL`` if a memory allocation failed
+      * ``ARK_ILL_INPUT`` if an argument has an illegal value.
+
+   .. versionadded:: x.y.z
+
+.. c:function:: int ARKStepCreateAdjointStepper(void* arkode_mem, N_Vector sf, SUNAdjointStepper* adj_stepper_ptr)
+
+   Creates a :c:type:`SUNAdjointStepper` object compatible with the provided ARKStep instance for
+   integrating the adjoint sensitivity system :eq:`ARKODE_DISCRETE_ADJOINT`.
+
+   :param arkode_mem: a pointer to the ARKStep memory block.
+   :param sf: the sensitivity vector holding the adjoint system terminal condition.
+      This must be an instance of the ManyVector ``N_Vector`` implementation with at
+      least one subvector (depending on if sensitivities to parameters should be computed).
+      The first subvector must be :math:`\partial g_y(y(t_f)) \in \mathbb{R}^N`. If sensitivities to parameters should be computed, then the second subvector must be :math:`g_p(y(t_f), p) \in \mathbb{R}^{N_s}`,
+      otherwise only one subvector should be provided.
+   :param adj_stepper_ptr: the newly created :c:type:`SUNAdjointStepper` object.
+
+   :return:
+      * ``ARK_SUCCESS`` if successful
+      * ``ARK_MEM_FAIL`` if a memory allocation failed
+      * ``ARK_ILL_INPUT`` if an argument has an illegal value.
+
+   .. versionadded:: x.y.z
+
+   .. note:: 
+
+      Currently only explicit methods are supported for ASA.
```

Comment:
I assume that non-identity mass matrices are also prohibited (even though ARKStep supports them for explicit methods)?  If that is the case, then I recommend a minor edit:
```suggestion
      Currently only explicit methods with identity mass matrices are supported for ASA.
```